client/core/core.go

```diff
@@ -958,6 +958,36 @@ func (c *Core) Login(pw []byte) ([]*db.Notification, error) {
 	return notes, nil
 }
 
+// Logout logs the user out
+func (c *Core) Logout() error {
+	c.connMtx.Lock()
+	defer c.connMtx.Unlock()
+
+	// Check active orders and lock wallets
+	for assetID := range c.User().Assets {
+		for _, dc := range c.conns {
+			if dc.hasOrders(assetID) {
+				return fmt.Errorf("cannot log out with active orders, wallet %s ", unbip(assetID))
+			}
+		}
+		wallet, found := c.wallet(assetID)
+		if found && wallet.connected() {
+			if err := wallet.Lock(); err != nil {
+				return err
+			}
+		}
+	}
+
+	// With no open orders for any of the dex connections, and all wallets locked,
+	// lock each dex account.
+	for _, dc := range c.conns {
+		dc.acct.lock()
+	}
+
+	c.refreshUser()
```

Comment:
Deadlock in `refreshUser>Exchanges` because `c.connMtx` is still locked from above.

Either unlock without the defer (before both returns and after all `c.conns` accesses), or use the defer stack for `c.refreshUser`.  That is, try putting `defer c.refreshUser()` **above** defer `c.connMtx.Unlock()`.  When `Logout` returns, it will unlock `c.connMtx` then run `c.refreshUser()`.  I think that is reasonable since `refreshUser` would be good even in the early return cases since wallets may have been locked.